pgmpy/factors/CPD.py

```diff
@@ -125,6 +125,79 @@ def __repr__(self):
+╘═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╛
+        >>> final_cpd.set_parents_order(['iq', 'intel', 'diff'])
+        >>> print(final_cpd)
+╒═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╤═════════╕
+│ diff    │ diff_0  │ diff_0  │ diff_0  │ diff_0  │ diff_1  │ diff_1  │ diff_1  │ diff_1  │ diff_2  │ diff_2  │ diff_2  │ diff_2  │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ intel   │ intel_0 │ intel_0 │ intel_1 │ intel_1 │ intel_0 │ intel_0 │ intel_1 │ intel_1 │ intel_0 │ intel_0 │ intel_1 │ intel_1 │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ iq      │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │ iq_0    │ iq_1    │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_0 │ 0.12    │ 0.5     │ 0.6     │ 0.5     │ 0.2     │ 0.4     │ 0.9     │ 0.9     │ 0.5     │ 0.8     │ 0.4     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_1 │ 0.12    │ 0.5     │ 0.6     │ 0.5     │ 0.2     │ 0.4     │ 0.9     │ 0.9     │ 0.5     │ 0.8     │ 0.4     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_2 │ 0.3     │ 0.7     │ 0.8     │ 0.7     │ 0.4     │ 0.2     │ 0.8     │ 0.8     │ 0.8     │ 0.8     │ 0.8     │ 0.4     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_3 │ 0.4     │ 0.8     │ 0.9     │ 0.8     │ 0.5     │ 0.1     │ 0.7     │ 0.6     │ 0.3     │ 0.8     │ 0.2     │ 0.8     │
+├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
+│ grade_4 │ 0.5     │ 0.9     │ 0.1     │ 0.9     │ 0.6     │ 0.8     │ 0.6     │ 0.5     │ 0.2     │ 0.8     │ 0.1     │ 0.8     │
+╘═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╧═════════╛
+        """
+        old_scope = self.scope()[::-1].copy()
+        new_cardinality = np.asarray(self.evidence_card)
+        new_order.append(self.variable)
+        list_a = [i for i in range(len(self.evidence_card))]
+        # list_b is to be used for cardinality mapping and column mapping.
+        list_b = [new_order[:].index(i) for i in old_scope]
+        new_cardinality[[x for x in list_b[:-1]]] = new_cardinality[[x for x in list_a]]
```

Comment:
@kislayabhi Simply doing `list_b[:-1]` would give the same result as `[x for x in list_b[:-1]]`.